client/src/pages/GrantAccessForm.js

```diff
@@ -0,0 +1,220 @@
+import { useState } from "react";
+import "./GrantAccessForm.css";
+import { post } from "../TenderClient";
+import Logo from "../assets/images/CTY-logo-rectangle.png";
+
+const GrantAccessForm = () => {
+	const [role, setRole] = useState("bidder");
+	const [buyerDetails, setBuyerDetails] = useState({
+		userType: "buyer",
+		company: "",
+		description: "",
+		address: "",
+		email: "",
+	});
+
+	const [bidderDetails, setBidderDetails] = useState({
+		userType: "bidder",
+		firstName: "",
+		lastName: "",
+		email: "",
+	});
+
+	const [resgisterStatus, setRegisterStatus] = useState("");
+	const [validationErrors, setValidationErrors] = useState([]);
+
+	async function postDetails(endpoint, data) {
+		try {
+			const res = await post(endpoint, data);
+
+			if (res && res.code !== "VALIDATION_ERROR") {
+				setRegisterStatus("Successfully registered.");
+				setValidationErrors([]);
+			} else {
+				setRegisterStatus("Registration failed!");
+				setValidationErrors(res.errors || []);
+			}
+		} catch (error) {
+			const { status, data } = error.response;
+
+			if (status === 403) {
+				setRegisterStatus("Access denied");
+				setValidationErrors(data.errors || []);
+			} else if (status === 500) {
+				setRegisterStatus("A server error occurred. Please try again later.");
+				setValidationErrors([]);
+			} else {
+				setRegisterStatus(error.response.data.code);
+				setValidationErrors(error.response.data.errors);
+			}
```

Comment:
```suggestion
			if (status === 400) {
				setRegisterStatus("Validation error");
				setValidationErrors(data.errors || []);
			} else {
				setRegisterStatus("A server error occurred. Please try again later.");
				setValidationErrors([]);
			}
```

If we get a 403, the client library will direct the user to the log-in page automatically